### PR TITLE
Documaster API: draft 14

### PR DESCRIPTION
**Pull request by ivaylomitrev**
Changes in the API documentation:

* Add recommendation for document title value

### CODE REVIEW

File: v1/model.md [hunk 1]
```diff
@@ -279,22 +279,22 @@ The Document resource is identified by `document` in URLs.
...
+| createdBy            | X    | string             | User name of the user who created the resource                                                                                   |
+| createdByUserId      | X    | string             | User ID of the user who created the resource                                                                                     |
+| updatedDate          | X    | timestamp          | Timestamp at which the resource was updated                                                                                      |
+| updatedBy            | X    | string             | User name of the user who last updated the resource                                                                              |
+| updatedByUserId      | X    | string             | User ID of the user who updated the resource                                                                                     |
+| title                | X    | string             | Title of the Document. Recommendation: set the title to the document version filename without the extension in your integration. |
```
Comment (phillipj) [inline v1/model.md:292]: I haven't been following along with recent discussions, you've probably discussed this a length already so excuse my stupid question: why is this recommended?

Knowing _why_ and aligning mindset is often a good motivator for following best practises. It's easier to ignore recommendations when the added benefit is not obvious :)

Comment (ivaylomitrev) [reply v1/model.md:292]: Good point!

This came as a suggestion from Arnstein. His point is that we should recommend to integrations what we already do as part of our internal processes (our own UI for example). Namely that we set the document title to the name of the uploaded file minus the extension. This improves consistency in the eyes of the user.

Comment (phillipj) [reply v1/model.md:292]: Gotcha, thx!

First food for thought that came to mind is making this best practise the default behaviour for everyone, as even better than aligning mindset as mentioned earlier, is avoiding humans having to think about following them at all 😄  Of course `title` could be explicitly set when that's needed for some reason.

Though there might be other aspects I'm not considering by having such default behaviour, that would be far worse than having to remember this best practise.

Comment (ivaylomitrev) [reply v1/model.md:292]: I generally agree with you, but I don't think I agree for this particular case.

First of all, I am not sure that we should be enforcing a best practice (as a default I mean). I've listed several potential pitfalls below.

Integrations might have a valid reason not to do it. For example their filename might be gibberish if it is produced by a third system. If they forget to override the default title, they'll get gibberish in the document title too :D 

Also, you can create an empty document and the default would then not kick in. Integrations that assumed that there's a default and don't need to set the title will then fail suddenly.

So... I'd be hesitant to make this the default behavior :D. I follow Arnstein's argument about being consistent, but for this particular recommendation, I'd vote to keep it at that - a recommendation.

In summary, following along my own brain dump in the sentences above, I think that this "default" recommendation is very conditional on the exact use case. I don't think it's general enough to warrant a default behavior in the application. I'm sure there'd be other though.

Comment (phillipj) [reply v1/model.md:292]: 👍 

Finding the balance and where sensible defaults can be used is challenging. Especially so when we don't have enough concrete insight to help us make informed decisions about what makes sense for the majority.

Regardless, this documentation update PR isn't the place or time to act on the above, still appreciate sharing some thoughts on the subject.
